src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2277,52 +2302,55 @@ void CodeGen::genBMI1Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
-        default:
+        case NI_BMI2_MultiplyNoFlags:
+        case NI_BMI2_X64_MultiplyNoFlags:
         {
-            unreached();
-            break;
-        }
-    }
-
-    genProduceReg(node);
-}
-
-//------------------------------------------------------------------------
-// genBMI2Intrinsic: Generates the code for a BMI2 hardware intrinsic node
-//
-// Arguments:
-//    node - The hardware intrinsic node
-//
-void CodeGen::genBMI2Intrinsic(GenTreeHWIntrinsic* node)
-{
-    NamedIntrinsic intrinsicId = node->gtHWIntrinsicId;
-    regNumber      targetReg   = node->gtRegNum;
-    GenTree*       op1         = node->gtGetOp1();
-    GenTree*       op2         = node->gtGetOp2();
-    var_types      baseType    = node->gtSIMDBaseType;
-    var_types      targetType  = node->TypeGet();
-    instruction    ins         = HWIntrinsicInfo::lookupIns(intrinsicId, targetType);
-    emitter*       emit        = getEmitter();
+            int numArgs = HWIntrinsicInfo::lookupNumArgs(node);
```

Comment:
nit: Having some newlines in the following block of code would help with readability. We've done such in most of the rest of the file.

Reply:
Fixed.